apps/web/src/server/applications/case/representations/representation/representation.middleware.js

```diff
@@ -0,0 +1,87 @@
+import { getCase, getRepresentation } from './representation.service.js';
+import { formatContactDetails } from './representation.utilities.js';
+
+/**
+ *
+ * @param {object} case
+ * @param {string} case.title
+ * @returns {{case: {projectName: string}}}
+ */
+const getCaseViewModel = ({ title }) => ({
+	case: {
+		projectName: title
+	}
+});
+
+/**
+ *
+ * @param {object} representation
+ * @param {Array<any>} representation.contacts
+ * @param {string} repType
+ * @returns {{representation: {contactDetails: object}}}
+ */
+const getRepresentationsViewModel = ({ contacts }, repType) => ({
+	representation: {
+		contactDetails: getContactDetails(contacts, repType)
+	}
+});
+
+/**
+ *
+ * @param {Array<any>} contacts
+ * @param {string} repType
+ * @returns {object}
+ */
+const getContactDetails = (contacts, repType) => {
+	const { represented, representative } = getContactDetailsByContactType(contacts);
+
+	switch (repType) {
+		case 'represented':
+			return represented;
+		case 'representative':
+			return representative;
+		default:
+			return formatContactDetails();
+	}
+};
+
+/**
+ *
+ * @param {Array<any>} contacts
+ * @returns {{represented: object, representative: object}}
+ */
+
+const getContactDetailsByContactType = (contacts) => {
+	const represented = formatContactDetails(contacts.find((element) => element.type !== 'AGENT'));
+	const representative = formatContactDetails(contacts.find((element) => element.type === 'AGENT'));
+
+	return {
+		represented,
+		representative
+	};
+};
+
+/**
+ * @param {import("express").Request} req
+ * @param {import("express").Response} res
+ * @param {import("express").NextFunction} next
+ */
+
+export const addRepresentationToLocals = async (req, res, next) => {
+	try {
+		const { params, query } = req;
+		const { caseId } = params;
+		const { repId, repType } = query;
+
+		res.locals = getCaseViewModel(await getCase(caseId));
```

Comment:
this should go to a key for the values -

Reply:
refactored to assign values to `res.locals.case` and `res.locals.contactDetails`